File: Object oriented programming (II sem)/OOP in cpp/Plant.cpp
#include"Plant.h"

Plant::Plant(string name, int strength, char icon, Position pos, World& world)
	: Organism(name, strength, PLANT_INITIATIVE, icon, pos, world) {
}

void Plant::action() {
	if (world.chanceOfBeeingTrue(PLANT_SPREADING_CHANCE)) {
		spread();
	}
}

void Plant::collision(Organism* other) {

}

void Plant::move(Position pos) {

}

void Plant::spread() {
	Position pos = findEmptyNeighbourField(position);
	if (pos.isCorrect()) {
		world.createNewOrganism(icon, pos);
		Announcer::addAnnounce(name + " spreaded");
	}
}

Plant::~Plant()
{
}

File: Object oriented programming (II sem)/OOP in java/Plant.java
import java.awt.*;

public class Plant extends Organism {
    public Plant(String name, int strength, Color color, Position position, World world) {
        super(name, strength, 0, color, position, world);
    }

    @Override
    public void action() {
        if (world.chanceOfBeingTrue(0.2f)) spread();
    }

    public void spread() {
        Position desiredPosition = this.getDesiredPosition(1);
        if (world.isEmpty(desiredPosition)) {
            world.addAnnounce(announcer.spreading(this));
            world.newLivingOrganism(this, desiredPosition);
        }
    }
}
